### ISSUE TITLE
TTS audio broken after recent refactoring

### ISSUE THREAD

**Issue by t2000** [openhab/openhab-core#2801]
When I tried to use the latest OH snapshot yesterday I noticed that my picotts isn't working with my squeezebox audio anymore.

In the logs I found:

```
2022-02-25 08:11:26.526 [WARN ] [core.voice.internal.VoiceManagerImpl] - Error saying 'My Message here': No compatible audio format found for TTS 'picotts' and sink 'squeezebox:squeezeboxplayer:piCorePlayer:SQ_Stefan'
```

I looked into the `VoiceManagerImpl` and found a change that was just done one day before my snapshot:

https://github.com/openhab/openhab-core/pull/2795

Apparently the refactoring in there is broken, because:

Before:

```
public static @Nullable AudioFormat getBestMatch(Set<AudioFormat> inputs, Set<AudioFormat> outputs) {
        AudioFormat preferredFormat = getPreferredFormat(inputs);
        for (AudioFormat output : outputs) {
            if (output.isCompatible(preferredFormat)) {
                return preferredFormat;
            } else {
                for (AudioFormat input : inputs) {
                    if (output.isCompatible(input)) {
                        return input;
                    }
                }
            }
        }
        return null;
    }
```

Now (since it uses the one in `AudioFormat.java`):

```
public static @Nullable AudioFormat getBestMatch(Set<AudioFormat> inputs, Set<AudioFormat> outputs) {
        AudioFormat preferredFormat = getPreferredFormat(inputs);
        if (preferredFormat != null) {
            for (AudioFormat output : outputs) {
                if (output.isCompatible(preferredFormat)) {
                    return preferredFormat;
                } else {
                    for (AudioFormat input : inputs) {
                        if (output.isCompatible(input)) {
                            return input;
                        }
                    }
                }
            }
        }
        return null;
    }
```
    
Which means that now there is a check for `if (preferredFormat != null)` around everything.
    
But its even a bit more complicated. The `VoiceManagerImpl` has (still!) its own `public static @Nullable AudioFormat getPreferredFormat(Set<AudioFormat> audioFormats)` which is DIFFERENT from the one in `AudioFormat`.

Apparently the one in `VoiceManagerImpl` is "older" since the last changes are > 2 years ago and the one in `AudioFormat` had some changes in the last months.

The `defaultFrequency` here:

https://github.com/openhab/openhab-core/blame/main/bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioFormat.java#L399

used to be 44100 instead of 16384.

Surprisingly it doesn't even get to the point to check if the frequencies are equal, because it fails at the endianess:

https://github.com/openhab/openhab-core/blame/main/bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioFormat.java#L291-L293
    
Since the `WAVE` format is defined here
https://github.com/openhab/openhab-core/blame/main/bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioFormat.java#L35-L36
without any endianess, I would have expected that this is a "generic format" that matches `true` AND `false`, however the aforementioned check in `Audioformat.isCompatible(@Nullable AudioFormat audioFormat)` doesn't do that and AFAIK hasn't done that before #2795 either.

So long story short: Please see the screenshot (had to wrap it in a zip since github didn't allow png or jpg) of my codecs. They were compatible before the recent changes, i.e. it IS possible to play the picotts wave output on my squeezebox sinks, but not it doesn't work anymore.

So there is something fishy and the only recent change is #2795 I think it has something to do with it, but i cannot pinpoint it out :(

[audio-formats.zip](https://github.com/openhab/openhab-core/files/8143261/audio-formats.zip)

**Comment by lolodomo** [comment]
Pinging @GiviMAD

**Comment by lolodomo** [comment]
Here is the unique format supported by the pico TTS binding: 
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.voice.picotts/src/main/java/org/openhab/voice/picotts/internal/PicoTTSService.java#L43
CONTAINER_WAVE, CODEC_PCM_SIGNED, false, 16, null, 16000L
